discv5/discv5-wire-test-vectors.md

```diff
@@ -1,339 +1,159 @@
-
-#### Input Parameters
-
-    secret-key: 0x02a77e3aa0c144ae7c0a3af73692b7d6e5b7a2fdc0eda16e8d5e6cb0d08e88dd04
-    node-id-A: 0xa448f24c6d18e575453db13171562b71999873db5b286df957af199ec94617f7
-    node-id-B: 0x885bba8dfeddd49855459df852ad5b63d13a3fae593f3f9fa7e317fd43651409
-    id-nonce: 0x0101010101010101010101010101010101010101010101010101010101010101
+This test vector checks the complete key derivation as used by the handshake.
 
-#### Expected Outputs
+    ephemeral-key = 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
+    dest-pubkey = 0x0317931e6e0840220642f230037d285d122bc59063221ef3226b1f403ddc69ca91
+    node-id-a = 0xaaaa8419e9f49d0083561b48287df592939a8d19947d8c0ef88f2a4856a69fbb
+    node-id-b = 0xbbbb9d047f0488c0b5a93c1c3f2d8bafc7c8ff337024a55434a0d0555de64db9
+    whoareyou.masking-iv = 0x01010101010101010101010101010101
+    whoareyou.id-nonce = 0x02020202020202020202020202020202
 
-The three keys, resulting from the HKDF-EXPAND function.
+The expected outputs, resulting from the HKDF-EXPAND function.
 
-    initiator-key: 0x238d8b50e4363cf603a48c6cc3542967
-    recipient-key: 0xbebc0183484f7e7ca2ac32e3d72c8891
-    auth-resp-key: 0xe987ad9e414d5b4f9bfe4ff1e52f2fae
+    initiator-key = 0xb10e94a89b34cfb87b65aa7f8902f40c
+    recipient-key = 0xce8db25ae599c9b2c4a9d60090c9efdd
 
-### Nonce Signing
+### ID Nonce Signing
 
-Nonce signatures should prefix the string `discovery-id-nonce` and post-fix the
-ephemeral key before taking the `sha256` hash of the `id-nonce`.
+This test vector checks the ID signature as used by the handshake.\
+The `static-key` is the secp256k1 private key used for signing.
 
-#### Input Parameters
+    static-key = 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
+    masking-iv = 0x01010101010101010101010101010101
+    authdata = 0x0102030405060708090a0b0c0d0e0f100000000000000000
+    ephemeral-pubkey = 0x039961e4c2356d61bedb83052c115d311acb3a96f5777296dcf297351130266231
+    node-id-B = 0xbbbb9d047f0488c0b5a93c1c3f2d8bafc7c8ff337024a55434a0d0555de64db9
 
-The `local-secret-key` is the raw secp256k1 private key used to sign the nonce.
+The expected output is the `id-signature`. You can also apply this test vector in reverse
+by verifying the signature against the inputs above.
 
-    id_nonce: 0xa77e3aa0c144ae7c0a3af73692b7d6e5b7a2fdc0eda16e8d5e6cb0d08e88dd04
-    ephemeral-key: 0x9961e4c2356d61bedb83052c115d311acb3a96f5777296dcf297351130266231503061ac4aaee666073d7e5bc2c80c3f5c5b500c1cb5fd0a76abbb6b675ad157
-    local-secret-key: 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
-
-#### Expected Outputs
-
-    id-nonce-sig: 0xc5036e702a79902ad8aa147dabfe3958b523fd6fa36cc78e2889b912d682d8d35fdea142e141f690736d86f50b39746ba2d2fc510b46f82ee08f08fd55d133a4
+    id-signature = 0xd82364cfffb18101355371de84ee0def3dca31191b9add79b21a14f4442b6df02dc26df6278f71c83d43645da13071881cacdb43b0aea1e256cdec73a73faf01
```

Comment:
@fjl I'm not able to reproduce this.  Can you maybe 1) check it against your implementation and 2) verify some of the intermediate values.

```
id-signature-input = 0x646973636f76657279207635206964656e746974792070726f6f66010101010101010101010101010101010102030405060708090a0b0c0d0e0f10039961e4c2356d61bedb83052c115d311acb3a96f5777296dcf297351130266231bbbb9d047f0488c0b5a93c1c3f2d8bafc7c8ff337024a55434a0d0555de64db9
message-hash: 0x762d64d468d019e113ba19dd68d4e11be16d989332e45402026871918b08d3f8
signature: 0x60e8ef785881dd89758b26346db5c0a779182d011519ecbb05aed14466db0a4a04b3853e3e34bec6517021c6e6d7beb293f3ba25305c80d094105e17849b0315
```

Reply:
I too cannot reproduce the value.
I can reproduce your values however, but only if I use just the `id-once` field in the `authdata`, which would be according to spec of course.